src/pymor/algorithms/timestepping.py

```diff
@@ -10,44 +10,363 @@
+        - pre-assembling stationary operators.
+
+        Note: internally calls :meth:_bootstrap, which has to be implemented by every time stepper.
+
+        Parameters
+        ----------
+        initial_data
+            The solution vector at `initial_time`.
+        operator
+            The |Operator| A.
+        rhs
+            The right-hand side F (either |VectorArray| of length 1 or |Operator| with
+            `source.dim == 1`). If `None`, zero right-hand side is assumed.
+        mass
+            The |Operator| M. If `None`, the identity operator is assumed.
+        mu
+            |Parameter values| for which `operator` and `rhs` are to be evaluated. The current
+            time is added to `mu` with key `t`.
+        reserve
+            If True and self.num_values is specified, pre-allocates the returned |VectorArray| for all time steps.
+
+        Returns
+        -------
+        t
+            Simply self.initial_time, to be used in loops as in :meth:`solve`
+        U
+            An empty |VectorArray| to hold the solution trajectory, to be used in loops as in :meth:`solve`.
+        data
+            A dictionary containing information required in :meth:`step` (like previously computed steps), as well
+            as cached data required by the time stepper (such as pre-assembled operators).
+        """
+        if not self.logging_disabled: # explicitly checking if logging is disabled saves some cpu cycles
+            self.logger.debug(f'bootstrapping (mu={mu}) ...')
+
+        assert isinstance(operator, Operator)
+        assert operator.source == operator.range
+
+        rhs = rhs or ZeroOperator(operator.source, NumpyVectorSpace(1))
+        assert isinstance(rhs, (Operator, VectorArray))
+        if isinstance(rhs, Operator):
+            assert rhs.source.dim == 1
+            assert rhs.range == operator.range
+        else:
+            assert rhs in operator.range
+            assert len(rhs) == 1
+            rhs = VectorArrayOperator(rhs)
+
+        mass = mass or IdentityOperator(operator.source)
+        assert isinstance(mass, Operator)
+        assert operator.source == mass.source == mass.range
+        assert not mass.parametric
+
+        if isinstance(initial_data, Operator):
+            assert initial_data.source.dim == 1
+            assert initial_data.range == operator.source
+        else:
+            assert initial_data in operator.source and len(initial_data) == 1
+            initial_data = VectorOperator(initial_data)
+
+        assert isinstance(self.steps, Number) and self.steps >= 1
+
+        if self.steps == 1:
+            # single-step method
+            U0, data = self._bootstrap(initial_data=initial_data, operator=operator, rhs=rhs, mass=mass, mu=mu)
+            assert len(U0) == 1
+            data['time_points'] = [self.initial_time,]
+            data['_previous_steps'] = U0
+            data['_previous_interpolation'] = lambda t: U0
+        else:
```

Comment:
Do we really need multi-step methods in pyMOR? (Also, you do not seem to implement any.) If yes, it might make more sense to handle this via inheritance?

Reply:
I would say we need those, yes (see other longer reply). They are on the way but not finished yet. We can see how useful they are once we have them. I am open to other interfaces, in particular regarding interpolation (see other reply). Regarding the need: now that I have to look at more complex problems (regarding time), it feels kind of awkward to not have more superior time stepping in pyMOR. In particular since this can be added generically given our interfaces and are then usable for a variety of pde solvers.